poetry/packages/package.py

```diff
@@ -40,7 +40,7 @@ def __init__(self, name, version, pretty_version=None):
         Creates a new in memory package.
         """
         self._pretty_name = name
-        self._name = canonicalize_name(name)
+        self.name = canonicalize_name(name)
```

Comment:
For now, it would be great to ensure this is set correctly as well. I'd suggest overridet setattr to ensure that the value is canonicalised.

Reply:
Thanks for taking a look at this change and really appreciate the feedback. 
When you say these changes should be submitted to the poetry/core what do you mean by this? 
Lastly happy to include this change in the MR.